### PR TITLE
aruha-602: return correct problem json response

### PR DESCRIPTION
**Pull request by adyach**
ARUHA-602: Server does not implement Problem JSON for auth errors

## Description
Changed error response according Problem Json, when the user is not authorised 

## Deployment Notes
Implement proper logic to support in Nakadi UI

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java [hunk 1]
```diff
@@ -116,6 +117,10 @@ public OAuth2Authentication loadAuthentication(final String accessToken) throws
                 } else {
                     return localService.loadAuthentication(accessToken);
                 }
+            } catch (OAuth2Exception e) {
+                throw e;
+            } catch (Exception e) {
```
Comment (antban) [inline src/main/java/org/zalando/nakadi/config/AuthenticationConfig.java]: Maybe RuntimeException?

File: src/main/java/org/zalando/nakadi/config/SecurityConfiguration.java [hunk 2]
```diff
@@ -77,35 +98,96 @@ public void configure(final HttpSecurity http) throws Exception {
...
+
+                return new ProblemResponse(Response.Status.fromStatusCode(oae.getHttpErrorCode()), oae.getMessage());
+            }
+
+            return new ProblemResponse(Response.Status.INTERNAL_SERVER_ERROR,
+                    "Unrecognized error happened in authentication path");
+        }
+    }
+
+    private static class ProblemResponse {
```
Comment (antban) [inline src/main/java/org/zalando/nakadi/config/SecurityConfiguration.java]: Why not to use `org.zalando.problem.DefaultProblem`?

File: src/main/java/org/zalando/nakadi/service/FeatureToggleServiceZk.java [hunk 3]
```diff
@@ -66,6 +66,8 @@ public void setFeature(final FeatureWrapper feature) {
             }
         } catch (final KeeperException.NoNodeException nne) {
             LOG.debug("Feature {} was already disabled", feature.getFeature().getId());
+        } catch (final KeeperException.NodeExistsException nne) {
+            LOG.debug("Feature {} was already enabled", feature.getFeature().getId());
```
Comment (antban) [inline src/main/java/org/zalando/nakadi/service/FeatureToggleServiceZk.java]: Nice! Finally Idempotent!

**Comment by antban** [comment]
:+1

**Comment by antban** [comment]
:+1:

**Comment by adyach** [comment]
:+1:

**Comment by adyach** [comment]
deploy validation please

**Pull request merged**